nsv13/code/modules/overmap/ai-skynet.dm

```diff
@@ -307,6 +307,50 @@ Adding tasks is easy! Just define a datum for it.
 		source_ship.hail(text, name, user.name, TRUE) // Let the crew on the source ship know an Outbound message was sent
 		hail(text, source_ship.name, user.name)
 
+/obj/structure/overmap/proc/try_deliver( mob/living/user, var/obj/machinery/computer/ship/dradis/cargo/console ) 
+	if( !isliving(user) )
+		return FALSE 
+	if( !allowed(user) ) //Only cargo auth'd personnel can make purchases.
+		to_chat(user, "<span class='warning'>Warning: You cannot open a communications channel without appropriate requisitions access registered to your ID card.</span>")
+		return FALSE
+
+	if ( console && console.linked_launcher )
+		var/obj/machinery/ship_weapon/torpedo_launcher/cargo/launcher = console.linked_launcher 
+		if ( launcher.chambered )
+			var/checkForLiving = recursive_locate( /mob/living, launcher.chambered )
+			
+			if ( !checkForLiving )
+				var/choice = input("Transfer cargo to [src]?", "Confirm delivery", "No") in list("Yes", "No")
+				if( choice == "Yes" ) 
+					var/obj/item/ship_weapon/ammunition/torpedo/freight/shipment = launcher.chambered 
+					receive_cargo( user, console, shipment )
+
+					// Fire the torpedo away to unload the launcher. 
+					// Without a weapon_type the projectile will not be animated 
+					launcher.fire( src, shots = 1 )
+				else 
```

Comment:
If you return here after a successful fire, you can just put a single
if(console.linked)
    try_hail(user, console.linked)
down at the end